modules/optimization/include/optimizationreporters/OptimizationReporter.h

```diff
@@ -9,17 +9,35 @@
 
 #pragma once
 
+#include "OptimizationDataHelper.h"
 #include "OptimizationReporterBase.h"
 
 /**
  * Computes gradient and contains reporters for communicating between optimizeSolve and subapps
  */
 class OptimizationReporter : public OptimizationReporterBase
 {
+private:
+  /// Data helper for generating objective value.
+  OptimizationDataHelper _opt_data;
+
```

Comment:
Move this to the bottom with the other private members

Reply:
I need `_opt_data` created before I initialize other public data. Should I just
make the other data pivate and place it before. We can an warning about the
order of initiliazing if I just move `_opt_data` to the bottom.